File: easy/3432. Count Partitions with Even Sum Difference/solution.java
// 3432. Count Partitions with Even Sum Difference
class Solution {
    public int countPartitions(int[] nums) {
        int total = 0;
        for (int x : nums) total += x;
        int left = 0, ans = 0;
        for (int i = 0; i < nums.length - 1; i++) {
            left += nums[i];
            int right = total - left;
            if (((left - right) & 1) == 0) ans++;
        }
        return ans;
    }
}

File: easy/3432. Count Partitions with Even Sum Difference/solution.rs
impl Solution {
    pub fn count_partitions(nums: Vec<i32>) -> i32 {
        let total: i32 = nums.iter().sum();
        let mut left = 0;
        let mut ans = 0;
        for i in 0..nums.len() - 1 {
            left += nums[i];
            let right = total - left;
            if ((left - right) & 1) == 0 {
                ans += 1;
            }
        }
        ans
    }
}